### PR TITLE
Make nav bar always visible and sticky

### PR DESCRIPTION
**Pull request by idradm**
## Links

* http://wikia-inc.atlassian.net/browse/DAT-4126

### CODE REVIEW

File: front/main/app/components/discussion-post-editor.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ export default DiscussionEditorComponent.extend({
 	 * @returns {boolean}
 	 */
 	isStickyBreakpointHeight() {
```
Comment (RafLeszczynski) [inline front/main/app/components/discussion-post-editor.js]: seams like a good candidate for computed property. @bkoval, @rafalkalinski, @kvas-damian was there any objections  for making this computed property?

**Comment by RafLeszczynski** [comment]
LGTM

**Comment by RafLeszczynski** [comment]
@dianafa please take a second look at these changes

File: front/main/app/mixins/theme.js [hunk 2]
```diff
@@ -69,7 +69,6 @@ export default Ember.Mixin.create(ColorUtilsMixin, {
 			return;
 		}
 
-		styles += `.discussions .border-theme-color {border-color: ${this.get('themeColors.color-buttons')};}`;
```
Comment (rafalkalinski) [inline front/main/app/mixins/theme.js]: Why is this line removed?

File: front/main/app/components/discussion-post-editor.js (hunk 1 above)
Comment (kvas-damian) [inline front/main/app/components/discussion-post-editor.js]: `window.pageYOffset` isn't observed by Ember, that's why we decided to not use `Ember.computed`. Probably it could be handled that way but I'm not sure if it's more readable  than current code.

File: front/main/app/mixins/theme.js (hunk 2 above)
Comment (kvas-damian) [inline front/main/app/mixins/theme.js]: Why this rule was removed?

File: front/main/app/styles/component/_discussion-editor.scss [hunk 3]
```diff
@@ -85,10 +85,6 @@ $editor-textarea-overflow-ios-hack: 30px;
 			transition: top 0.5s;
 			width: $sticky-editor-width;
 			z-index: 1;
-
-			&.pinned-top {
```
Comment (kvas-damian) [inline front/main/app/styles/component/_discussion-editor.scss]: Please make sure removing this block doesn't break discussion editor UI.

File: front/main/app/styles/component/_discussion-header.scss [hunk 4]
```diff
@@ -10,11 +10,10 @@ $site-header-height: 50px;
 	position: fixed;
 	top: 0;
 	width: 100%;
-	z-index: $z-8; // drop-downs from top-bar component (which is $z-7) display over the discussion header
+	z-index: $z-6; // drop-downs from top-bar component (which is $z-7) display over the discussion header
```
Comment (kvas-damian) [inline front/main/app/styles/component/_discussion-header.scss]: Is this comment still up-to-date?

File: front/main/app/templates/components/discussion-editor.hbs [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <label class="editor-label">{{i18n labelText ns='discussion'}}</label>
-<div class="editor-container {{if isSticky 'sticky'}} {{unless siteHeadPinned pinnedClassName}}">
+<div class="editor-container {{if isSticky 'sticky'}} {{pinnedClassName}}">
```
Comment (rafalkalinski) [inline front/main/app/templates/components/discussion-editor.hbs]: If siteHeadPinned is now always true, then beeing consistent with previous condition, pinnedClassName should never be here anymore.

Comment (idradm) [inline front/main/app/templates/components/discussion-editor.hbs]: `siteHeadPinned` is always true, however, this code was using headroom functionality to turn on/off class for two components `discussion-post-editor` and `discussion-reply-editor`.

This could be removed for the first component, although would break the second one, as it's used here: https://github.com/Wikia/mercury/blob/92e7a559eb7f199f7be5e130a59ee6ef69135336/front/main/app/components/discussion-reply-editor.js#L6

File: front/main/app/styles/component/_discussion-header.scss (hunk 4 above)
Comment (idradm) [inline front/main/app/styles/component/_discussion-header.scss]: Except the `display over...` part, will remove it.

File: front/main/app/styles/component/_discussion-editor.scss (hunk 3 above)
Comment (idradm) [inline front/main/app/styles/component/_discussion-editor.scss]: This css was used to move editor to top on scroll down when the bar was hidden. Which is no longer a case.

File: front/main/app/mixins/theme.js (hunk 2 above)
Comment (idradm) [inline front/main/app/mixins/theme.js]: We will be adding `fandom bar` bellow the nav, therefore, it was removed. Additionally, it collided with `smart-banner`.

Comment (kvas-damian) [inline front/main/app/mixins/theme.js]: This class wasn't only used on nav. See: https://github.com/Wikia/mercury/blob/04255056fa7ac165346b2be41d53dd61c02e7032/front/main/app/templates/components/discussion-sort.hbs#L2-L2

Comment (kvas-damian) [inline front/main/app/mixins/theme.js]: :+1:

File: front/main/app/components/discussion-post-editor.js (hunk 1 above)
Comment (RafLeszczynski) [inline front/main/app/components/discussion-post-editor.js]: let's leave it as it is for now but this is something we could consider changing it the future

File: front/main/app/styles/component/_site-head.scss [hunk 6]
```diff
@@ -44,19 +44,8 @@ $site-head-height: 50; //unitless
...
-
-.headroom-not-top {
-	&.pinned {
-		transform: none;
-	}
-
-	&.un-pinned {
-		transform: translateY(-100%);
-	}
+.application-wrapper {
```
Comment (RafLeszczynski) [inline front/main/app/styles/component/_site-head.scss]: shouldn't this be put inside application-wrapper styles?

**Comment by RafLeszczynski** [comment]
LGTM if no more remarks from Social Team. @kvas-damian @rafalkalinski are you ok with that?

File: front/main/app/components/site-head.js [hunk 7]
```diff
@@ -1,73 +1,63 @@
...
+		 */
+		trackWordmarkClick() {
+			track({
+				action: trackActions.click,
+				category: 'wordmark'
+			});
+		}
+	},
+
+	didRender() {
```
Comment (bkoval) [inline front/main/app/components/site-head.js]: How about a small, ebautiful jsdoc for this method? :)

Comment (RafLeszczynski) [inline front/main/app/components/site-head.js]: this is ember component lifecycle hook

**Comment by kvas-damian** [comment]
It LGTM although please check if it doesn't break discussion functionality ;)

File: front/main/app/styles/component/_site-head.scss (hunk 6 above)
Comment (idradm) [inline front/main/app/styles/component/_site-head.scss]: This should happen only if `site-head` is enabled, that's the reason it's here.

File: front/main/app/components/site-head.js (hunk 7 above)
Comment (bkoval) [inline front/main/app/components/site-head.js]: What a method above has :)

**Comment by RafLeszczynski** [comment]
@kvas-damian we will do regression testing

**Comment by RafLeszczynski** [comment]
@idradm please address remaining comments and move to QA

**Comment by bkoval** [comment]
LGTM!

File: front/main/app/mixins/theme.js (hunk 2 above)
Comment (rafalkalinski) [inline front/main/app/mixins/theme.js]: Sounds good,but it probably should be removed from here: https://github.com/Wikia/mercury/pull/2400/files#diff-3c1465af792c139083130ff66c2d96aeR5
